src/iuse_actor.cpp

```diff
@@ -917,3 +917,619 @@ bool extended_firestarter_actor::can_use( const player* p, const item* it, bool
 
     return true;
 }
+
+salvage_actor::~salvage_actor() {}
+
+void salvage_actor::load( JsonObject &obj )
+{
+    moves_per_part = obj.get_int( "moves_per_part", 25 );
+    if( obj.has_array( "material_whitelist" ) ) {
+        JsonArray jarr = obj.get_array( "material_whitelist" );
+        for( int i = 0; i < (int)jarr.size(); ++i ) {
```

Comment:
Better: `for( size_t i = 0; i < jarr.size(); ... `

Or use the "canonical" way (used nearly everywhere else):
```C++
while( jarr.has_more() ) {
    const auto v = jarr.next_string();
```

Or use `JsonObject::get_string_array`:
```C++
material_whitelist = obj.get_string_array( "material_whitelist" );
```

Also: I agree with Rivet on this PR.